### ISSUE TITLE
Regarding the loss function curve issue of uvcganv2 on my own dataset?

### ISSUE THREAD

**Issue by zcm-1231** [LS4GAN/uvcgan2#47]
This is the loss function curve on my own dataset.
> The specific dataset configuration is as follows: Domain A (1603 images) and Domain B (1611 images)
> The specific hyperparameter configuration is` "--lambda-gp 1.0"`, `"--lambda-cycle 5.0"`, and `"--lr-gen 5e-5"`.
> The FID was calculated. The results for 500 rounds are (A fid:49 B fid:102); the results for 800 rounds are (A 60 B99).
Now I have a few questions to ask the  #author
1. Regarding the parameters during pre-training of the model, 'model' can be set to either 'simple-autoencoder' or 'autoencoder'. Do these different parameters have a significant impact on different tasks?
2. Is the loss function curve above a normal training process?
3. I have tried setting` "--lambda-cycle 10.0"` to make the generated image from A->B consistent with the original image, so I reduced it to 5.0. When `"--lambda-gp 0.01" `is used, the result is also as above. Therefore, this configuration yields the best effect. So, how should I adjust my hyperparameters next? Or should I not make any adjustments?
Looking forward to your reply. This will be very helpful to me. Thank you!

<img width="2684" height="1769" alt="Image" src="https://github.com/user-attachments/assets/661645c0-c238-4561-9480-7b25822e6085" />

**Comment by usert5432** [comment]
Hi @zcm-1231,

Thank you for your question.

Looking at your loss curves, the discriminator loss is going to zero while the generator loss increases. This probably means the discriminator has become too strong relative to the generator.

Usually, there are a few things that can help with that:
- increasing gradient penalty (`lambda-gp`)
- decreasing discriminator learning rate
- increasing generator learning rate (`lr-gen`)

I would try either increasing lambda-gp to 10 or increasing lr-gen to 0.0001 or 0.0002 as the first step.

If the above does not help, we can try investigating more -- that would require inspecting the dataset itself.

Regarding your specific questions:
1. The pre-training model choice probably won't fix this D/G imbalance issue. The difference between `autoencoder` and `simple-autoencoder` is just in what input data format they expect.
2. These curves are a bit abnormal - the discriminator seems too strong relative to the generator.

Please let me know if you have any other questions.

**Comment by zcm-1231** [comment]
Thank you for your reply. I have taken your advice and adjusted `lambda-gp 10.0 `and `lr-gen 0.0001`. However, the FID from domain A to domain B increases with each epoch, while the FID from B to A decreases. It may be that there is a problem with inconsistent mappings in my dataset.

I also tried downloading models for other tasks from the paper and plotting loss curves, but I'm not sure where the differences lie between my loss curves and those in the original paper. Could you suggest how I should adjust the hyperparameters moving forward? I asked the AI, and it told me that if I only care about the transfer from A to B, I can try reducing `lambda-cycle`.

I hope you can give me some advice.Below is the loss function curve I plotted.

<img width="1600" height="1060" alt="Image" src="https://github.com/user-attachments/assets/1a520429-766f-4a37-b1d1-456aff0a8253" />

<img width="2683" height="1769" alt="Image" src="https://github.com/user-attachments/assets/fcc22094-b78e-4fce-a269-1b277a2d3382" />

<img width="2683" height="1769" alt="Image" src="https://github.com/user-attachments/assets/d42ee789-f5f2-499d-9ebc-9c356980c20b" />

**Comment by usert5432** [comment]
Hi @zcm-1231,

Unfortunately, I do not have a good idea what the problem may be here from the loss curves alone.

However, I think I saw a similar behavior before. It happened with datasets where domains are not very aligned (e.g. scale difference between domain A and B). If the dataset is not private, would it be possible for you to share a few samples and their translations to see what is going on there (compare middle of the training vs end phase)?
